### PR TITLE
Create TGSA.def

### PR DESCRIPTION
**Pull request by smujiang**

### CODE REVIEW

File: definitions/TGSA.def [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    # will be installed and removed in post step
+    cp ./src/Singularity_gpu_fix.sh $SINGULARITY_ROOTFS/
+
+%environment
+        export PATH=/usr/local/TGSA:/usr/local/cuda/bin:$PATH
+        export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
+        export CUDA_INSTALL_DIR=/usr/local/cuda
+        export CUDA_HOME=/usr/local/cuda
+        export MODEL_DIR=/usr/local/TGSA_model
+        echo "export MODEL_DIR=/usr/local/TGSA_model" >> ~/.bashrc
```
Comment (wilke) [inline definitions/TGSA.def:18]: Why do you need to add this to bashrc

File: definitions/TGSA.def [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    # install python modules and model prerequisites
+    pip install git+https://github.com/ECP-CANDLE/candle_lib@develop
+
+    #pip install CPython==3.8.0
+    pip install torch-geometric==1.7.1
+    pip install torch_cluster==1.5.9
+    pip install torch_scatter==2.0.9
+    pip install torch_sparse==0.6.12
+    pip install torch_spline_conv==1.2.1
+    #wget https://data.pyg.org/whl/torch-1.9.0%2Bcu102/torch_cluster-1.5.9-cp38-cp38-linux_x86_64.whl
```
Comment (wilke) [inline definitions/TGSA.def:74]: Can we remove these lines if not needed?

File: definitions/TGSA.def [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    #rm ./*.whl
+
+    #conda env create -f TGSA/singularity/TGSA.yml
+
+    # setup model and executables
+    cd /usr/local/
+    git clone https://github.com/smujiang/TGSA.git
+
+    cp TGSA/data/enterez_NCBI_to_hugo_gene_symbol_march_2019.txt /candle_data_dir/
```
Comment (wilke) [inline definitions/TGSA.def:107]: This will fail when mounting into /candle_data_dir, the config file won't be there.

**Review by wilke — CHANGES_REQUESTED** [review]

File: definitions/TGSA.def [hunk 4]
```diff
@@ -0,0 +1,115 @@
+%Bootstrap: docker
```
Comment (wilke) [inline definitions/TGSA.def:1]: Remove %

**Comment by wilke** [comment]
Does not build, tested from https://github.com/smujiang/Singularity.git branch patch-1 :

`singularity build --fakeroot build/TGSA.def definitions/TGSA.def
FATAL:   Unable to build from definitions/TGSA.def: while parsing definition: definitions/TGSA.def: section %Bootstrap: docker: could not be split into section name and body`

**Comment by wilke** [comment]
After commenting out the first two lines and rebuilding with the command above:

```
INFO:    Running post scriptlet
+ apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
E: gnupg, gnupg2 and gnupg1 do not seem to be installed, but one of them is required for this operation
FATAL:   While performing build: while running engine: exit status 255
```

**Comment by brettin** [comment]
It appears this might be needed: 
apt-get update && apt-get install -y gnupg2

File: definitions/TGSA.def (hunk 1 above)
Comment (smujiang) [reply definitions/TGSA.def:18]: previously, I had a CUDA driver issue. These could be removed.

**Comment by wilke** [comment]
Merged into TGSA branch, testing

**Comment by wilke** [comment]
Closing this PR - code in TGSA branch - definition file needs updates.

**Pull request closed without merging**